types/modAction.ts

```diff
@@ -0,0 +1,71 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { DocumentData } from 'firebase/firestore';
+import 'react-native-get-random-values';
+import { v4 as uuidv4 } from 'uuid';
+import { LazyObject, User } from '../types';
+
+export class ModAction extends LazyObject {
+  public userModerated?: User;
+  public mod?: User;
```

Comment:
I'd appreciate the comments in the same style as the other lazy objects, the ones saying which ones are required and which ones are optional. 

Also, could we rename `mod` to `moderator`? It seems like `mod` can be a noun or a verb, and `mod` interpreted as a verb might mean the thing reported, not the person doing the moderating

One last thing, maybe I'm missing something - why don't we have a ref to the moderated content here?

Reply:
oh I see, i'm stupid, if the content is deleted then what would the reference be to. :p